src/components/QuizApp.tsx

```diff
@@ -17,14 +17,25 @@ const QuizApp: React.FC = () => {
   const [selectedAnswer, setSelectedAnswer] = useState<number | null>(null);
   const [isSubmitted, setIsSubmitted] = useState<boolean>(false);
   const [score, setScore] = useState<number>(0);
-  const [questions, setQuestion] = useState<QuizQuestion[]>([]);
+  const [questions, setQuestions] = useState<QuizQuestion[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false); // Track question loading
 
   const startQuiz = () => {
     setCurrentQuestionIndex(0);
     setScore(0);
     setSelectedAnswer(null);
     setIsSubmitted(false);
-    setQuestion(retrieveQuizQuestions());
+    setIsLoading(true);
+
+    getQuestions()
+      .then((fetchedQuestions: QuizQuestion[]) => {
+        setQuestions(fetchedQuestions);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.error(err);
+        setIsLoading(false); // Stop loading even on error
+      });
```

Comment:
```suggestion
    getQuestions()
      .then((fetchedQuestions: QuizQuestion[]) => {
        setQuestions(fetchedQuestions);
      })
      .catch((err) => {
        console.error(err);
      })
      .finally(() => {
        setIsLoading(false); 
      });
```

Also, do we want to show some sort of error state?

And, could be a future iteration, but there are some libraries that can handle loading, errors, and even caching quite nicely.
ex: https://tanstack.com/query/latest

Reply:
I think an error state makes sense. I will take a little bit more time to familiarize myself with the library that you suggested, but I will definitely do it in the next iteration. Thanks for pointing it out!

Reply:
Update: TanStack Query is now used to handle loading and errors